src/rp2040/hardware_structs/include/hardware/structs/timer.h

```diff
@@ -8,26 +10,87 @@
 #define _HARDWARE_STRUCTS_TIMER_H
 
 #include "hardware/address_mapped.h"
-#include "hardware/platform_defs.h"
 #include "hardware/regs/timer.h"
 
-#define NUM_TIMERS 4
```

Comment:
This `#define` has disappeared, and is causing the build to fail.
Although a `git grep` shows that `src/host/pico_platform/include/hardware/platform_defs.h` contains `#define NUM_TIMERS 4u`; so perhaps `src/rp2040/hardware_regs/include/hardware/platform_defs.h` also ought to contain a similar define?

Reply:
Indeed. It can always be added back, but I found that including it in the structs was a bit odd given how everything else is in platform defs.

Also, there are a bunch of errors now due to access rights and fields being const. I'm just going off the SVD file, so if it's complaining it means the info in the SVD isn't correct.

Reply:
Yeah, there _is_ already a whole bunch of other `NUM_foo` defines in https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2040/hardware_regs/include/hardware/platform_defs.h :slightly_smiling_face: 

> Also, there are a bunch of errors now due to access rights and fields being const.

Yeah, there's ~~errors~~warnings like:
```
/opt/actions-runner/_work/pico-sdk/pico-sdk/src/rp2_common/hardware_gpio/gpio.c: In function 'gpio_irq_handler':
/opt/actions-runner/_work/pico-sdk/pico-sdk/src/rp2_common/hardware_gpio/gpio.c:145:32: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
[  6%] Building C object test/pico_stdlib_test/CMakeFiles/pico_stdlib_test.dir/__/__/src/common/pico_sync/mutex.c.obj
         io_rw_32 *status_reg = &irq_ctrl_base->ints[gpio / 8];
```
which is coming from https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_gpio/gpio.c#L145 but the RP2040 datasheet says that `IO_BANK0: PROC0_INTS0` is indeed entirely "RO", so _perhaps_ it would make sense to change `src/rp2_common/hardware_gpio/gpio.c` to do
```c
io_ro_32 *status_reg = &irq_ctrl_base->ints[gpio / 8];
```
instead? But that's **definitely** a question for Luke or Graham :wink: